Neos.Flow/Classes/Security/Context.php

```diff
@@ -783,8 +784,13 @@ protected function updateTokens(array $tokens)
             $token->updateCredentials($this->request);
         }
 
-        $sessionDataContainer = $this->objectManager->get(SessionDataContainer::class);
-        $sessionDataContainer->setSecurityTokens(array_merge($this->inactiveTokens, $this->activeTokens));
+        $tokensForSession = array_filter(array_merge($this->inactiveTokens, $this->activeTokens), static function(TokenInterface $token) {
+            return !$token instanceof SessionlessTokenInterface;
+        });
+        if ($tokensForSession !== []) {
```

Comment:
Just to make sure: if there are already tokens in the session data from previously but now no more tokens exist, the session data will no longer be overwrittten with that empty list. Is this potentially an issue?

Reply:
Good catch! I was thinking about this one, too - and then discarded the thought.. But maybe it's safer to set the empty array in this case?

What I was wondering (and didn't test yet): Would that start a session even though I don't want that if I only have sessionless tokens?

Reply:
That's funny, because that is the other taught I had, that I discarded, because a session is started anyway in different places (CSRF for example) 😄

Guess we'd need to have a "getSessionDataContainerIfSessionExists" method.